basics_of_sst-core/tutorial/Example00.md

```diff
@@ -0,0 +1,568 @@
+
+The `setup()` and `finish()` methods are also required to interface with the SST core functions.  `setup()` is called after the component constructor but before the simulation starts.  As such, it's a good place to put any component initialization that can't be done in the constructor.  `finish()` is called after the simulation ends and it a good place to perform any necessary cleanup associated with the component.
+
+As for the clock event handler, when configuring the component in the previously mentioned simulation configuration file, the component will be assigned a clock rate.  On each clock tick, the clock event handler is called.  This makes the clock event handler a good place to put code that needs to be executed on a periodic basis.
+
+The component is described to the SST core using the SST_ELI macros.
+
+```
+            // Shared documentation macros. 
+            //
+            SST_ELI_DOCUMENT_PARAMS(
+                { "debug", "Debug location:  (0: NONE, 1: STDOUT, 2: STDERR, 3: FILE)", "0" },
+                { "clock", "Component clock rate", "1GHz" },
+                { "clockTicks", "Number of times the handler is called before ending.", "10" }
+            )
+            
+            SST_ELI_REGISTER_COMPONENT(
+                ExampleComponent,                       // Class name
+                "example",                              // Library name (the *.so)
+                "ExampleComponent",                     // Name used to reference the component.  This can be
+                                                        // whatever you want it to be and will be referenced
+                                                        // in the python configuration file.
+                SST_ELI_ELEMENT_VERSION( 1, 0, 0 ),     // Version number
+                "Clock element example",                // Description                 
+                COMPONENT_CATEGORY_UNCATEGORIZED        // Component category
+            )
+```
+
+The `SST_ELI_DOCUMENT_PARAMS` macro defines the component parameters and their default values.  The `SST_ELI_REGISTER_COMPONENT` macro registers the component with the ELI database, making it available for use by the simulation configuration file.
+
+Finally, a few private variables that will be used to track state within the component are defined.
+
+```
+        private:
+            // Member variables for this example.
+            //
+            SST::Output logger_;        // For displaying log messages.
+            uint64_t componentId_;      // SST supplied component id.
+            uint64_t clockTicks_;       // Maximum number of clock ticks.
+            uint64_t clockTickCount_;   // Clock ticks counter.
+```
+
+## Implementing Component Functionality
+
+To implement the component functionality, use the editor of your choice and create a new file.  Enter the code shown below and save it as `ExampleComponent.cc`.
+
+```
+#include "ExampleComponent.h"
```

Comment:
All .cc files need to include sst_config.h before anything else. No .h file should include that header. See http://sst-simulator.org/SSTPages/SSTDeveloperCodingStandards/.